docs/api/js/utils.md

```diff
@@ -1,68 +1,563 @@
+
+```ts
+export async function isMessageSignatureCorrect(
+    provider: Provider,
+    address: string,
+    message: ethers.Bytes | string,
+    signature: SignatureLike
+): Promise<boolean> {
+    const msgHash = ethers.utils.hashMessage(message);
+    return await isSignatureCorrect(provider, address, msgHash, signature);
+}
+```
+
+
+### `isSignatureCorrect`
+
+Called from [`isMessageSignatureCorrect`](#ismessagesignaturecorrect) and [`isTypedDataSignatureCorrect`](#istypeddatasignaturecorrect). Returns true if account abstraction EIP712 signature is correct.
+
+```ts
+async function isSignatureCorrect(
+    provider: Provider,
+    address: string,
+    msgHash: string,
+    signature: SignatureLike
+): Promise<boolean> {
+    let isContractAccount = false;
+
+    const code = await provider.getCode(address);
+    isContractAccount = ethers.utils.arrayify(code).length != 0;
+
+    if (!isContractAccount) {
+        return isECDSASignatureCorrect(address, msgHash, signature);
+    } else {
+        return await isEIP1271SignatureCorrect(provider, address, msgHash, signature);
+    }
+}
+```
+
+### `isTypedDataSignatureCorrect`
+
+Returns true if account abstraction EIP712 signature is correct.
+
+```ts
+export async function isTypedDataSignatureCorrect(
+    provider: Provider,
+    address: string,
+    domain: TypedDataDomain,
+    types: Record<string, Array<TypedDataField>>,
+    value: Record<string, any>,
+    signature: SignatureLike
+): Promise<boolean> {
+    const msgHash = ethers.utils._TypedDataEncoder.hash(domain, types, value);
+    return await isSignatureCorrect(provider, address, msgHash, signature);
+}
+```
+
+### `parseTransaction`
+
+Common parsing transaction function used by internal teams. 
+
+Please see the [utilities library definition](https://github.com/matter-labs/zksync-2-dev/blob/94701bd2fbc590f733346934cfbccae08fc62f1a/sdk/zksync-web3.js/src/utils.ts) for more info.
+
+### `serialize`
+
+Common serialize function used by internal teams. 
+
+Please see the [utilities library definition](https://github.com/matter-labs/zksync-2-dev/blob/94701bd2fbc590f733346934cfbccae08fc62f1a/sdk/zksync-web3.js/src/utils.ts) for more info.
+
+### `serializeTransaction`
+
+Common serialize transaction function used by internal teams. 
+
+Please see the [utilities library definition](https://github.com/matter-labs/zksync-2-dev/blob/94701bd2fbc590f733346934cfbccae08fc62f1a/sdk/zksync-web3.js/src/utils.ts) for more info.
+
+### `undoL1ToL2Alias`
+
+Converts the `msg.sender` viewed on L2 to the address that submitted a transaction to the inbox on L1.
+
+See also [`applyl1tol2alias`](#applyl1tol2alias).
 
-export const RECOMMENDED_GAS_LIMIT = {
-    DEPOSIT: 600_000,
-    EXECUTE: 620_000,
-    ERC20_APPROVE: 50_000
-};
+```ts
+export function undoL1ToL2Alias(address: string): string {
+    return ethers.utils.hexlify(ethers.BigNumber.from(address).sub(L1_TO_L2_ALIAS_OFFSET));
+}
```

Comment:
up-to-date function version:
```suggestion
```ts
export function undoL1ToL2Alias(address: string): string {
    let result = ethers.BigNumber.from(address).sub(L1_TO_L2_ALIAS_OFFSET);
    if (result.lt(BigNumber.from(0))) {
        result = result.add(ADDRESS_MODULO);
    }

    return ethers.utils.hexlify(result);
}
```